### DISCUSSION TITLE
Anisotropic and Rate independent phase field fracture

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20005] [Q&A Modules: General] [answered]
Hello MOOSE Experts

I want to solve rate-independent and anisotropic phase-field fracture. Can you please provide me with close examples to solve the same?
I shall be very thankful.

Best
Avtar

**Comment by hugary1995** [comment]
Hi Avtar,

There are plenty of examples in `modules/combined/test/tests/phase_field_fracture/`.

Alternatively, there is an app for phase field fracture: [https://hugary1995.github.io/raccoon/](https://hugary1995.github.io/raccoon/). I haven't put any anisotropic fracture models in yet (which should be trivial BTW), but if you are interested, we could discuss.

Thanks,
Gary

**Comment by avtarsinghh1991** [comment, marked as answer]
Hello Gary

Thank you very much. I am new to MOOSE. I have checked the  

`modules/combined/test/tests/phase_field_fracture/ `

in which Allen-Cahn equation is solved. However, It is time-dependent and I can't able to find anything where I can block the time derivative term.

For example
https://mooseframework.inl.gov/source/materials/ComputeLinearElasticPFFractureStress.html

```
 #This input uses PhaseField-Nonconserved Action to add phase field fracture bulk rate kernels
 [Mesh]
   [gen]
     type = GeneratedMeshGenerator
     dim = 2
     nx = 40
     ny = 20
     ymax = 0.5
   []
   [./noncrack]
     type = BoundingBoxNodeSetGenerator
     new_boundary = noncrack
     bottom_left = '0.5 0 0'
     top_right = '1 0 0'
     input = gen
   [../]
 []
 
 [GlobalParams]
   displacements = 'disp_x disp_y'
 []
 
 [AuxVariables]
   [./strain_yy]
     family = MONOMIAL
     order = CONSTANT
   [../]
 []
 
 [Modules]
   [./TensorMechanics]
     [./Master]
       [./All]
         add_variables = true
         strain = FINITE
         planar_formulation = PLANE_STRAIN
         additional_generate_output = 'stress_yy'
         strain_base_name = uncracked
       [../]
     [../]
   [../]
   [./PhaseField]
     [./Nonconserved]
       [./c]
         free_energy = E_el
         kappa = kappa_op
         mobility = L
       [../]
     [../]
   [../]
 []
 
 [Kernels]
   [./solid_x]
     type = PhaseFieldFractureMechanicsOffDiag
     variable = disp_x
     component = 0
     c = c
  [../]
   [./solid_y]
     type = PhaseFieldFractureMechanicsOffDiag
     variable = disp_y
     component = 1
     c = c
   [../]
   [./off_disp]
     type = AllenCahnElasticEnergyOffDiag
     variable = c
     displacements = 'disp_x disp_y'
     mob_name = L
   [../]
 []
 
 [AuxKernels]
   [./strain_yy]
     type = RankTwoAux
     variable = strain_yy
     rank_two_tensor = uncracked_mechanical_strain
     index_i = 1
     index_j = 1
     execute_on = TIMESTEP_END
   [../]
 []
 
 [BCs]
   [./ydisp]
     type = FunctionDirichletBC
     variable = disp_y
     boundary = top
     function = 't'
   [../]
   [./yfix]
     type = DirichletBC
     variable = disp_y
     boundary = noncrack
     value = 0
   [../]
   [./xfix]
     type = DirichletBC
     variable = disp_x
     boundary = right
     value = 0
   [../]
 []
 
 [Materials]
   [./pfbulkmat]
     type = GenericConstantMaterial
     prop_names = 'gc_prop l visco'
     prop_values = '1e-3 0.05 1e-4'
   [../]
   [./elasticity_tensor]
     type = ComputeElasticityTensor
     C_ijkl = '120.0 80.0'
     fill_method = symmetric_isotropic
     base_name = uncracked
   [../]
   [./elastic]
     type = ComputeFiniteStrainElasticStress
     base_name = uncracked
   [../]
   [./cracked_stress]
     type = ComputeCrackedStress
     c = c
     kdamage = 1e-5
     F_name = E_el
     use_current_history_variable = true
     uncracked_base_name = uncracked
     finite_strain_model = true
   [../]
 []
 
 [Postprocessors]
   [./av_stress_yy]
     type = ElementAverageValue
     variable = stress_yy
   [../]
   [./av_strain_yy]
     type = SideAverageValue
     variable = disp_y
     boundary = top
   [../]
 []
 
 [Preconditioning]
   [./smp]
     type = SMP
     full = true
   [../]
 []
 
 [Executioner]
   type = Transient
 
   solve_type = PJFNK
   petsc_options_iname = '-pc_type -pc_factor_mat_solving_package'
   petsc_options_value = 'lu superlu_dist'
 
   nl_rel_tol = 1e-8
   l_tol = 1e-4
   l_max_its = 100
   nl_max_its = 10
 
   dt = 3e-5
   num_steps = 2
 []
 
 [Outputs]
   exodus = true
 []
```


I find this example as most suitable for me. How can I make this Allen-Cahn into a rate-independent one?
Please help.

I shall be very thankful.

Best
Avtar

**Reply by hugary1995** [reply]
Another input file called `crack2d_iso_wo_time.i` is time-independent I believe.

**Reply by hugary1995** [reply]
@dschwen Is it worth adding an option in `NonconservedAction` to turn off the time derivative kernel?

**Reply by avtarsinghh1991** [reply]
Thank you @hugary1995 .

I will check the examples you suggested. I need to put anisotropy as well. So I will try to write kernel for the same.

Best
Avtar

**Reply by ngrilli** [reply]
Dear @asingh-mit  
Anisotropy is already implemented here:
https://github.com/idaholab/moose/blob/next/modules/combined/test/tests/phase_field_fracture/crack2d_aniso_cleavage_plane.i
Best Regards,
Nicolò

**Comment by avtarsinghh1991** [comment]
Hello @ngrilli 

Thank you very much. Actually, I want to implement this for the polycrystalline system. So, I think I need to change the following source code in order to read the cleavage plane values for individual grains from the input file.

https://mooseframework.inl.gov/docs/doxygen/modules/ACInterfaceCleavageFracture_8C_source.html

Best
Avtar
